purchase_line_refund_to_invoice_qty/models/purchase.py

```diff
@@ -0,0 +1,65 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    qty_refunded_not_invoiceable = fields.Float(
+        compute="_compute_qty_refunded_not_invoiceable",
+        string="Quantity Refunded Not Invoiceable",
+        digits="Product Unit of Measure",
+    )
+
+    @api.depends(
+        "invoice_lines.move_id.state",
+        "invoice_lines.quantity",
+        "invoice_lines.purchase_qty_to_reinvoice",
+        "order_id.state",
+        "qty_received"
+    )
+    def _compute_qty_invoiced(self):
+        res = super()._compute_qty_invoiced()
+        for line in self:
+            qty_invoiced = line.qty_invoiced
+            for invoice_line in line.invoice_lines:
+                if (
+                    invoice_line.move_id.state != "cancel"
+                    and invoice_line.move_id.move_type == "in_refund"
+                    and not invoice_line.purchase_qty_to_reinvoice
+                ):
+                    qty_invoiced += invoice_line.product_uom_id._compute_quantity(
+                        invoice_line.quantity, line.product_uom
+                    )
+
+            line.qty_invoiced = qty_invoiced
+            if line.order_id.state in ['purchase', 'done']:
+                if line.product_id.purchase_method == 'purchase':
+                    line.qty_to_invoice = line.product_qty - line.qty_invoiced
+                else:
+                    line.qty_to_invoice = line.qty_received - line.qty_invoiced
+            else:
+                line.qty_to_invoice = 0
```

Comment:
This block could be skipped if we notice that `line.qty_invoiced` haven't changed, but it's not relevant IMO